src/main/java/org/cyclops/integrateddynamics/core/recipe/type/RecipeFacadeSqueeze.java

```
@@ -0,0 +1,56 @@
+package org.cyclops.integrateddynamics.core.recipe.type;
+
+import org.cyclops.cyclopscore.recipe.ItemStackFromIngredient;
+import org.cyclops.integrateddynamics.RegistryEntries;
+import org.cyclops.integrateddynamics.item.ItemFacade;
+
+import com.mojang.datafixers.util.Either;
+
+import net.minecraft.core.NonNullList;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.Container;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.item.crafting.RecipeSerializer;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.fluids.FluidStack;
+
+public class RecipeFacadeSqueeze extends RecipeSqueezer {
+
+    private Ingredient inputIngredient;
+    private IngredientChance facadeItemChance = new IngredientChance(Either.<ItemStack, ItemStackFromIngredient>left(new ItemStack(RegistryEntries.ITEM_FACADE)), 1.0f);
+
+    public RecipeFacadeSqueeze(ResourceLocation id, Ingredient inputIngredient, NonNullList<IngredientChance> outputItems, FluidStack outputFluid) {
+        super(id, inputIngredient, outputItems, outputFluid);
+        this.inputIngredient = inputIngredient;
+    }
+
+    public NonNullList<IngredientChance> assemble(ItemStack inputItem) {
+
+        if (!(inputItem.getItem() instanceof ItemFacade)) {
```

It might be better to just make these cases illegal in the serializer.

---

I can def move it to the serializer, would you prefer it silently fail to load, or log a warning?